Assets/Scripts/Managers/BattleQueue.cs

```diff
@@ -280,11 +410,426 @@ public List<ActionClass> GetList()
+            /*            while (BattleQueueInstance.actionQueue.GetList().Count != 0)
+                        {
+                            BattleQueueInstance.actionQueue.GetList().Remove(BattleQueueInstance.actionQueue.GetList()[0]);
+                        }
+                        // alternatively
+                        BattleQueueInstance.actionQueue.GetList().Clear();*/
+        }
+
+        // only called inside Insert() iff Insert() method of SortedArray is about to return true; not that it returns it; is called before the method ends. // ASTER2
+        /* When a player plays a card against an enemy, the fastest action the enemy has will get promoted to the player�s speed. Even if the Enemy Attack is 
+                targeting a different target. CASE 1
+
+            If the enemy�s speed is higher than the player�s speed. The player�s card gets promoted up instead. However, it only promotes player cards 
+                that are targeting THEM, and will not promote attacks targeting different enemies. CASE 2 */
+        // if the above two are conditions are not germane create a new wrapper.
+        public void InsertPlayerActionIntoWrappers(ActionClass playerAct)
+        {
+            // implementation is based on the fact that we are not checking already clashing entities. i.e. first enemy action is doled out. 
+            // do perform check for availability
+
+            // since this is only called if there is a successful insertion
+            if (BattleQueue.BattleQueueInstance.roundStart)
+            {
+                EnemyActionsTransformation(); // implemented therein roundStart = false too;
+            }
+            if (playerAct != null)
+            {
+
+                foreach (Wrapper curWrapper in wrappers)
+                {
+                    if (curWrapper.PlayerAction == null && curWrapper.EnemyAction != null) // if the wrapper is half-empty
+                    {
+                        if (playerAct.Target == curWrapper.EnemyAction.Origin && curWrapper.EnemyAction.Speed > playerAct.Speed) // CASE 2; the card in wrapper is by the targeted enemy and ITS speed is greater; relying on reference equality 
+                        {
+                            curWrapper.PlayerAction = playerAct;
+                            curWrapper.Update();
+                            SortWrappers();
+                            // DisplayWrapperArray();
+                            return;
+                        }
+                        else if (playerAct.Target == curWrapper.EnemyAction.Origin)// CASE 1; 
+                        {
+                            curWrapper.PlayerAction = playerAct;
+                            curWrapper.Update();
+                            SortWrappers();
+                            // DisplayWrapperArray();
+                            return;
+                        }
```

Comment:
Second case here logically seems redundant to me. If your proposed case #2 here is true, the case 1 will always be true as well. Since both paths are the same, the top case must be redundant